### ISSUE TITLE
[FEATURE] Add training script that uses our data 

### ISSUE THREAD

**Issue by jmj23** [coreadai/pytorch-image-models#1]
**Is your feature request related to a problem? Please describe.**
We would like to train using our custom data. 

**Describe the solution you'd like**
The sample data is in CSV format. However, we should be able to swap out the loaded CSV data with other data sources. We will eventually get the training data from mongodb. The data will ultimately be loaded as a dict in the format `{image_path: image_label}`, so we just have to swap out the code that loads the CSV into a dict.
We would also like to retain access to all features of this repo so we may experiment with different training parameters.

**Comment by mansikataria** [comment]
1. Added PyTorch Dataset(https://github.com/coreadai/pytorch-image-models/blob/ISSUE-1_Add_training_script_that_uses_our_data/timm/data/dataset.py#L152) to deal with Dict (as discussed on Upwork).
2. The Dataset identifies unique class labels and encodes them
3. Here is a training script: https://github.com/coreadai/pytorch-image-models/blob/ISSUE-1_Add_training_script_that_uses_our_data/train_coreadai_img_classification.py
4. It expects a data/train.csv file.


The training script will accept a bunch of command-line arguments, I'll prepare a guide for it, but here are some important arguments:
data/ --dataset=coaiclass --model=efficientnetv2_m --input-size 1 256 256 --num_classes=x
